internal/utils/sarif/sarif.go

```diff
@@ -25,14 +27,26 @@ func SarifLevelToSeverity(level string) string {
 	return severity
 }
 
+func WithCoverage(coverage []sarif.SarifCoverage) CodeSummaryOptions {
+	res := func(s *json_schemas.TestSummary) *json_schemas.TestSummary {
+		for _, v := range coverage {
+			s.Artifacts = s.Artifacts + v.Files
+		}
+		return s
+	}
+	return res
+}
+
 // Iterate through the sarif data and create a summary out of it.
-func CreateCodeSummary(input *sarif.SarifDocument) *json_schemas.TestSummary {
+func CreateCodeSummary(input *sarif.SarifDocument, options ...CodeSummaryOptions) *json_schemas.TestSummary {
 	if input == nil {
 		return nil
 	}
 
 	summary := json_schemas.NewTestSummary(summaryType)
 	resultMap := map[string]*json_schemas.TestSummaryResult{}
+
+	summary.Artifacts = 0
```

Comment:
Suggestion: How about moving the default value to `json_schemas.NewTestSummary()`?